protocol_test.go

```diff
@@ -97,16 +98,26 @@ func TestLongCallCancel(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	var longResult int
 	var err error
-	wait := runInBg(func() error {
+
+	type result struct {
+		res int
+		err error
+	}
+	resultCh := make(chan result)
+	runInBg(func() error {
 		longResult, err = cli.LongCall(ctx)
-		return err
+		resultCh <- result{longResult, err}
+		longResult, err = cli.LongCallResult(context.Background())
+		resultCh <- result{longResult, err}
+		return nil
 	})
+	time.Sleep(1 * time.Millisecond)
```

Comment:
Does this test depend on the sleep working? If so, then that means the test is buggy. :/ You can't rely on the Sleep causing some other particular goroutine to run.

Can you rewrite the test without requiring sleeps?